dc backgrounds and job post (#580)
deltachat/deltachat-pages: _posts/2022-12-12-ui-development.md

Our Android app is a 2019-fork of Signal's Java app,
diverged in some areas because we orient ourselves
around the interfaces of Telegram and WhatsApp.
The Android app remains our "flagship" app that is available in various stores,
among them Google Play and F-Droid.
Our iOS app is a self-development using Uikit and Swift bindings to core.
It is the youngest sibling in our offerings and also needs more love.

For both Desktop and mobile development we offer 80-120 hours-per-month contracts.
Payment is not bad but not Silicon-Valley level either.
Read more about [our funding sources here](https://delta.chat/en/help#how-are-delta-chat-developments-funded).
We are happy to discuss situational arrangements and adaptations.
Contributing and collaborating happens remote mostly
but several of us tend to meet every few months
in Freiburg (black forest), elsewhere in Germany (Berlin, Hamburg, Leipzig…),
and from time to time in more remote locations like Kyiv or Havanna.
We typically invite and converse with folks from interesting or befriended projects,
collocated in decentralization, internet freedom, and queer fusion spaces.

> Several of us avoid flying and some are involved in FFF or adjacent activities.

Reviewer: ```suggestion
Several of us avoid flying and some are involved in the Future of Freedom Foundation or adjacent activities.
```

If we gather somewhere remote then usually for weeks rather than days,
and without seeking much publicity.
We try to avoid constant busyness and urgency as it prevents enjoyable collaboration.

> We aim to arrange ourselves [reliably and empathically](community-standards)

Author: changed link and link text. thanks.

Delta Chat app developments are a usability and UX-driven effort and
for this we have experienced UX researchers with us.
Security considerations play a day-to-day role, however, the "most secure protocol"
is useless if few are able to understand or use it.
"Usable security" itself remains an evasive concept
if we can not facilitate its implementation in real-life apps.
Between usability, security, and implementation considerations
there is no natural hierarchy: each constrains and influences the others.
In 2023, we want to perform perform systematic user and field testing around security topics,
in part following the often-quoted usable security study ["When Signal hits the fan"](https://dx.doi.org/10.14722/eurousec.2016.23012).

### Easing onboarding with "Automatic Accounts"

Using a pre-existing email account is one of Delta Chat's core features, but also 
happens to be an impediment for many when getting started with Delta Chat.
Supplying the e-mail address and password is enough for autoconfiguration to
quickly succeed, but not without problems and although plenty of e-mail providers work
with Delta Chat, several have unnecessary limitations and complications.

> Many users do not have a good e-mail address and can not easily get one.

Reviewer: What is a 'good' e-mail address?

Reviewer 2: Agree. Is it about the email providers tech and reputation? Or about the adress itself? ("good email adress" = beautiful email adress?)

Reviewer 2: Also, may be link to our previous UX reports (eg from Kyiv) https://delta.chat/en/2018-12-19-needfinding

Reviewer 2: and I would also may be point at the relativity of risk, threat modelling and real-life approach in usable security (as in our paper with Francesca Musiani https://www.erudit.org/fr/revues/im/2018-n32-im04484/1058473ar/)

Author: i stroke the sentence. this removes the confusion.  We say already e-mail providcers have several limitations

Rust enables large-scale collaboration between developers
on a wide variety of platforms without runtime-overheads.
C and C++ could never deliver this at similar scale,
however foundational they have been and are for today's state of things.
Our Rust core implements all networking, message processing, encryption,
chat and contact persistence, which offers a documented UX-oriented API for UIs and bots.
Core is licensed under the MPL and thus more permissive
than our user interface developments which are largely licensed under the GPL.

Our apps and bots use Rust-core bindings for Java, Swift, TypeScript and Python.
While our mobile apps use the long-evolved CFFI (C-Foreign-Function-Interface),
our Desktop app introduces JSON-RPC (JSON Remote-Procedure-Calls),
directly talking to core without a C-layer.
New asynchronous python bindings doing away with all CFFI were started this week
with first bots being ported to it.

### Bots, bots, bots

For mailing-lists, SuperGroups, Mastodon, screenshots, downloading, and other fun experiments,
bots are being deployed for practical and entertaining contexts.
But this post is getting long and twisted enough already…

> Let's get to the maybe most important point of this whole post…

Reviewer: I understand this might be a style, but I generally remove things like this as it slows down 'getting to the point'.

I put bots in it's own paragraph so people who scan heading know it's there and because it's not directly related to Rust.

Author: i modified it but think it's important to recognize that rushing away after a sentence about bots just so is weird -- but writing much more about bots when we are actually looking for UX/UI devs  is as well.

---

> title: Want to help Delta Chat become more beautiful and need money?

Reviewer: ```suggestion
title: Want to earn money improving Delta Chat UI?
```
I would avoid saying 'beautiful' as it focuses on the cosmetic aspect, which is important also, but ideally this person would do more than make-up (i.e. consider the design more holistically).

Reviewer: Could also try a 'calling all UX/UI designers…' vibe

- **resilient**: Internationally dispersed families and friends stay in contact,
  even when WhatsApp, Signal, VPNs, and Tor fail.

- **sovereign**: Members of a monastery (ethically barred from using "remote" accounts)
  run their own e-mail server with a chat-messenger that they accept as a gift (yay Free and Open Source Software, or FOSS).

- **interoperable**: Anyone can chat even if the recipient doesn't have the app —
  it works with any e-mail address.

### Why don't we promote our project more?

Since the beginnings of Delta Chat in a small city north of Hamburg almost five years ago,
we have posted little about these wonderful and diverse ways people use the technology.
We usually prefer to share completed app releases instead of pre-announcing or
trying to create hype by boasting about users and growth.
Avoiding popularity is not merely born out of humility, but also realism as
we can only address so much feedback and suggestions.
If we are over capacity, feature requests and ideas turn into pressure and stress.

Luckily we have a wonderful testing community with key contributors
sorting out issues with users and discussing with developers.

> One from Austria kept reminding us about "changed-network" connectivity problems

Reviewer: I think this could have its own heading of 'community' or 'contributors' instead of being joined with 'promotion'.

Reviewer 2: I would add may be 

"As users are intrinsically operating and existing in "multi-tool" and multi-app settings, we have never pursued a goal to replace other messaging apps".

Author: i put it to a different section but skipped the "not replacing other messaging apps" part for now :)

> One from Austria kept reminding us about "changed-network" connectivity problems

Reviewer: ```suggestion
One from Austria kept reminding us about specific connectivity problems
```
Not sure if the technical specificity adds much. I'd take it out.

even if they rarely occur and are hard to re-produce:
the issues were fixed this week, which increased resiliency for everyone.
Did you know that most Delta Chat features are first tested on a Caribbean island?
Some were originally developed in the DeltaLab app:
a friendly fork of Delta Chat Android only available in the Cuban app store.

### Extending decentralized chat with web apps

[Apps shared in a Chat](https://webxdc.org) were released in mid-2022.
Dozens of grassroots-ported games are played daily in chat groups and through mailing lists.
Collaborative tooling apps (polls, checklists, calendars, editors) are evolving.
No permission from us, or any app store, is needed to create and distribute your apps.
Programming newcomers succeeded in delivering working usable apps via a zip file
containing an `index.html`, some assets and JS/Typescript/CSS as preferred.

> One of our more experienced and playful developers recently whipped up a chat app

Reviewer: Might want to say why it makes sense. I wasn't able to figure it out.

Author: added some more explanation

> that runs inside a chat group — makes more sense than you might think.

Reviewer: Might want to say why it makes sense. I wasn't able to figure it out.

Our Poll example app is 6 kilobytes and, as with all chat-shared web apps,
automatically operates offline-first and end-to-end encrypted
without any server deployments (and thus without logins or GDPR/consent screens).

> Experts and fediverse friends agree:
> web tech combined with decentralized chat (aka "webxdc") is a rare jewel worth refining.
> Some friends and long-term followers hailed our approach as a "decentralized super-app".
> Can you guess from which island these developments were inspired and co-driven?

Reviewer: ```suggestion
Experts and fediverse friends agree:
web tech combined with decentralized chat (aka "webxdc") is a rare jewel worth refining and has been hailed as a "decentralized super-app".
```

I would be more specific about the 'experts'. Not necessary to name them directly, but enough reference that the reader doesn't have to trust us. Has a kind of 'the science says that…' vibe the way it is. It also sounds a bit weak when combining 'experts' and 'fediverse friends': maybe add another group in there, or rephrase it all as one group.

The 'which island' question was already repeated in the previous section, with the answer right after, so I'd leave it out, or if it's important to acknowledge then say something like 'these developments were in large part driven by our community in Cuba'.

Reviewer 2: Agree with the need to open up the "experts" black box (e.g. privacy researchers and developers and fediverse friends...). 

Also agree that explicitly mentioning Cuba would be good.

Author: i fixed it to be more specific.

### Why don't we promote our project more?

Since the beginnings of Delta Chat in a small city north of Hamburg almost five years ago,
we have posted little about these wonderful and diverse ways people use the technology.
We usually prefer to share completed app releases instead of pre-announcing or
trying to create hype by boasting about users and growth.
Avoiding popularity is not merely born out of humility, but also realism as
we can only address so much feedback and suggestions.
If we are over capacity, feature requests and ideas turn into pressure and stress.

Luckily we have a wonderful testing community with key contributors
sorting out issues with users and discussing with developers.
One from Austria kept reminding us about "changed-network" connectivity problems
even if they rarely occur and are hard to re-produce:
the issues were fixed this week, which increased resiliency for everyone.
Did you know that most Delta Chat features are first tested on a Caribbean island?
Some were originally developed in the DeltaLab app:
a friendly fork of Delta Chat Android only available in the Cuban app store.

### Extending decentralized chat with web apps

> [Apps shared in a Chat](https://webxdc.org) were released in mid-2022.

Reviewer: Different names for the same thing here:
- on the webxdc site, it's: web apps shared in a chat
- the first link is: 'Apps shared in a Chat'
- later it's 'chat-shared'
- and at the end it's 'aka "webxdc"'.

I think if we will use the term webxdc at all, then we can put it at the start (either replacing or in addition to '[web] apps shared in a chat') and re-use it later, as we would with an acronym. Or if we will start calling it private apps, then just use that. But I would pick one name and stick to it.

Author: i streamlined a little but not sure we can not fully resolve it through this post.   i do think "webxdc" is important for to keep as a term.  In the UIs i find "private apps" actually too long (eg on android).
